snuba/migrations/backfill_errors.py

```diff
@@ -6,462 +6,169 @@
+        print("Error data was found")
     except IndexError:
-        last_migrated = datetime.utcnow()
+        ts = datetime.utcnow()
 
-    src_columns = ", ".join(
-        [cast(str, escape_identifier(field)) for field in Event.field_names()]
-    )
+    timestamp = get_monday(ts.date())
 
-    events_migrated = 0
+    print(f"Starting migration from {format_date(timestamp)}")
 
     while True:
-        if isinstance(last_migrated, MigratedEvent):
-            event_conditions = f"""
-            AND (
-                (
-                    timestamp = {escape_literal(last_migrated.timestamp)}
-                    AND (
-                        project_id < {escape_literal(last_migrated.project_id)}
-                        OR (
-                            project_id = {escape_literal(last_migrated.project_id)}
-                            AND event_id < {escape_literal(last_migrated.event_id)}
-                        )
-                    )
-                )
-                OR (
-                    timestamp < {escape_literal(last_migrated.timestamp)}
-                )
-            )
-            AND timestamp >= {escape_literal(last_migrated.timestamp - MAX_BATCH_WINDOW)}
-            """
-        else:
-            event_conditions = f"""
-            AND timestamp >= {escape_literal(last_migrated - MAX_BATCH_WINDOW)}
-            AND timestamp <= {escape_literal(last_migrated)}
-            """
-
-        raw_events = clickhouse.execute(
-            f"""
-            SELECT {src_columns} FROM {events_table_name}
-            WHERE type != 'transaction'
-            AND NOT deleted
-            {event_conditions}
-            ORDER BY timestamp DESC, project_id DESC, event_id DESC
-            LIMIT {MAX_BATCH_SIZE}
-            """
+        where = f"toMonday(timestamp) = toDate('{format_date(timestamp)}') AND (deleted = 0)"
+
+        operation = InsertIntoSelect(
+            storage_set=StorageSetKey.EVENTS,
+            dest_table_name=errors_table_name,
+            dest_columns=[c[0] for c in COLUMNS],
+            src_table_name=events_table_name,
+            src_columns=[c[1] for c in COLUMNS],
+            prewhere="type != 'transaction'",
+            where=where,
         )
+        clickhouse.execute(operation.format_sql())
 
-        if len(raw_events) == 0:
-            if isinstance(last_migrated, datetime):
-                if last_migrated - MAX_BATCH_WINDOW < BEGINNING_OF_TIME:
-                    print(f"Done. Migrated {events_migrated} events")
-                    return
-                else:
-                    last_migrated -= MAX_BATCH_WINDOW
-            else:
-                last_migrated = last_migrated.timestamp - MAX_BATCH_WINDOW
-        else:
-            (event_id, project, timestamp,) = raw_events[-1][0:3]
-            last_migrated = MigratedEvent(event_id, project, timestamp)
+        print(f"Migrated {format_date(timestamp)}.")
```

Comment:
Could you pass a logger to this instead of using bare print statements ? When we run it from console we would have the same result, and when we use it into a migration it lets the migration framework control the output.